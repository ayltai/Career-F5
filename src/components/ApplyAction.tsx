import { RocketLaunch as RocketLaunchIcon, } from '@mui/icons-material';
import { Button, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

export const ApplyAction = ({
    onClick,
} : {
    onClick : () => void,
}) => {
    const { t, } = useTranslation();

    return (
        <Button
            color='primary'
            variant='contained'
            startIcon={
                <RocketLaunchIcon sx={{
                    color : 'white',
                }} />
            }
            onClick={onClick}>
            <Typography
                color='white'
                variant='button'>
                {t('action_apply')}
            </Typography>
        </Button>
    );
};
