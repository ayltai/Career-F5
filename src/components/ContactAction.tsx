import { WhatsApp as WhatsAppIcon, } from '@mui/icons-material';
import { Button, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

const handleClick = () => {
    if (process.env.REACT_APP_WHATSAPP_PHONE_NUMBER) {
        const newWindow = window.open(`https://wa.me/${process.env.REACT_APP_WHATSAPP_PHONE_NUMBER}`, '_blank', 'noopener,noreferrer');
        if (newWindow) newWindow.opener = null;
    }
};

export const ContactAction = () => {
    const { t, } = useTranslation();

    return (
        <Button
            variant='contained'
            color='secondary'
            startIcon={
                <WhatsAppIcon sx={{
                    color : 'white',
                }} />
            }
            onClick={handleClick}>
            <Typography
                color='white'
                variant='button'>
                {t('whatsapp')}
            </Typography>
        </Button>
    );
};
