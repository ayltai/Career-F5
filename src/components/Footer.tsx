import { Box, Container, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

export const Footer = () => {
    const { t, } = useTranslation();

    return (
        <Box
            sx={{
                backgroundColor: 'black',
            }}
            width='100%'>
            <Container maxWidth='lg'>
                <Box
                    paddingY={4}
                    textAlign='center'>
                    <Typography
                        color='white'
                        variant='overline'>
                        {t('copyright')}
                    </Typography>
                </Box>
            </Container>
        </Box>
    );
};
