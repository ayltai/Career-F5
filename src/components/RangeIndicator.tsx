import { Box, Slider, Stack, Typography, } from '@mui/material';
import React, { ReactNode, } from 'react';

import type { Profession, } from '../@types';

export const RangeIndicator = (props : Profession & {
    color?  : string,
    min     : number,
    max     : number,
    format? : (value : number) => ReactNode,
}) => (
    <Stack
        paddingY={2}
        direction='column'>
        <Stack
            display='flex'
            direction='row'>
            <Box flexGrow={1}>
                <Typography
                    color={props.color}
                    variant='body1'>
                    {props.name}
                </Typography>
            </Box>
            <Box flexGrow={0}>
                <Typography
                    color={props.color}
                    variant='body1'>
                    {props.format ? props.format(props.salaries.avg) : props.salaries.avg}
                </Typography>
            </Box>
        </Stack>
        <Slider
            sx={{
                height : 8,
            }}
            color='primary'
            min={props.min}
            max={props.max}
            value={[
                props.salaries.min,
                props.salaries.max,
            ]}
            valueLabelDisplay='auto'
            valueLabelFormat={props.format} />
    </Stack>
);
