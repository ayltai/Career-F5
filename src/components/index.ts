export { ApplyAction, } from './ApplyAction';
export { ApplyDialog, } from './ApplyDialog';
export { ContactAction, } from './ContactAction';
export { Footer, } from './Footer';
export { PaymentPlan, } from './PaymentPlan';
export { RangeIndicator, } from './RangeIndicator';
export { TopBar, } from './TopBar';
