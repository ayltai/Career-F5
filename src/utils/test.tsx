import { ThemeProvider, } from '@mui/material';
import { configureStore, PreloadedState, Store, } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { render, } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { preferencesReducer, } from '../states';
import { AppTheme, } from '../styles';

export const createMatchMedia = (width : number) => (query : string) => ({
    matches             : mediaQuery.match(query, {
        width,
    }),
    media               : '',
    onchange            : null,
    addListener         : () => {},
    removeListener      : () => {},
    addEventListener    : () => {},
    removeEventListener : () => {},
    dispatchEvent       : () => false,
});

const createStore = (preloadedState? : Record<string, any>) => configureStore({
    reducer : {
        preferences : preferencesReducer,
    },
    preloadedState,
});

const defaultStore = createStore();

type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?  : PreloadedState<RootState>,
    store?           : Store<RootState>,
    [ rest : string] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <ThemeProvider theme={AppTheme('en')}>
                {children}
            </ThemeProvider>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
