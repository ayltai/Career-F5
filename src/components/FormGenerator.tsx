import { Button, Card, CardActions, CardContent, Typography, useMediaQuery, } from '@mui/material';
import { GridContainer, widgets, } from '@ui-schema/ds-material';
import { createEmptyStore, createOrderedMap, injectPluginStack, isInvalid, relT, StoreSchemaType, storeUpdater, TranslatorContext, UIMetaProvider, UIStoreProvider, } from '@ui-schema/ui-schema';
import React, { useCallback, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import { handleError, } from '../utils';

const GridStack = injectPluginStack(GridContainer);

export type FormGeneratorProps = {
    title?       : string,
    description? : string,
    definition   : object,
    onSubmit     : (data : Record<string, string>) => Promise<boolean>,
};

export const FormGenerator = ({
    elevated = true,
    title,
    description,
    definition,
    onSubmit,
} : FormGeneratorProps & {
    elevated? : boolean,
}) => {
    const storeSchema = createOrderedMap({
        type : 'object',
        ...definition,
    });

    const [ store,        setStore,        ] = useState(() => createEmptyStore(storeSchema.get('type')));
    const [ showValidity, setShowValidity, ] = useState(false);
    const [ isSubmitting, setIsSubmitting, ] = useState(false);

    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, } = useTranslation();

    const handleChange = useCallback((actions : any) => setStore(storeUpdater(actions)), [ setStore,]);

    const handleSubmit = () => {
        setShowValidity(true);

        if (!isInvalid(store.getValidity())) {
            setIsSubmitting(true);

            onSubmit(store.valuesToJS() as Record<string, string>).then(success => {
                if (success) {
                    setStore(createEmptyStore(storeSchema.get('type')));

                    setShowValidity(false);
                }
            }).then(() => setIsSubmitting(false)).catch(handleError);
        }
    };

    const translate = (text : string, context? : TranslatorContext, schema? : StoreSchemaType) => {
        const schemaT = relT(schema, context);

        return schemaT ? schemaT : t(text);
    };

    return (
        <Card
            sx={{
                borderRadius : 4,
                padding      : desktopMode ? 4 : 1,
            }}
            elevation={elevated ? 4 : 0}>
            <CardContent>
                {title && (
                    <Typography
                        gutterBottom
                        variant='h5'
                        fontWeight='bold'>
                        {t(title)}
                    </Typography>
                )}
                {description && (
                    <Typography
                        gutterBottom
                        variant='body1'>
                        {t(description)}
                    </Typography>
                )}
                <UIMetaProvider
                    t={translate}
                    widgets={widgets}>
                    <UIStoreProvider
                        showValidity={showValidity}
                        store={store}
                        onChange={handleChange}>
                        <GridStack
                            isRoot
                            schema={storeSchema} />
                    </UIStoreProvider>
                </UIMetaProvider>
            </CardContent>
            <CardActions>
                <Button
                    sx={{
                        marginX : 1,
                    }}
                    disabled={isSubmitting || !!isInvalid(store.getValidity())}
                    variant='contained'
                    onClick={handleSubmit}>
                    <Typography
                        color='white'
                        variant='button'>
                        {t(isSubmitting ? 'action_submitting' : 'action_submit')}
                    </Typography>
                </Button>
            </CardActions>
        </Card>
    );
};
