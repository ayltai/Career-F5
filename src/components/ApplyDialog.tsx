import { Close as CloseIcon, } from '@mui/icons-material';
import { Box, Dialog, DialogContent, DialogContentText, DialogTitle, IconButton, Typography, useMediaQuery, useTheme, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

import { FormGenerator, FormGeneratorProps, } from './FormGenerator';

export const ApplyDialog = ({
    open = false,
    onClose,
    title,
    description,
    definition,
    onSubmit,
} : FormGeneratorProps & {
    open?    : boolean,
    onClose? : () => void,
}) => {
    const theme = useTheme();

    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, } = useTranslation();

    return (
        <Dialog
            open={open}
            onClose={onClose}>
            <DialogTitle>
                <Box
                    paddingX={desktopMode ? 4 : 1}
                    paddingTop={desktopMode ? 4 : 1}>
                    <Typography variant='h5'>
                        {title}
                    </Typography>
                    <IconButton
                        sx={{
                            position : 'absolute',
                            right    : desktopMode ? 16 : 8,
                            top      : desktopMode ? 16 : 8,
                            color    : theme.palette.text.secondary,
                        }}
                        onClick={onClose}>
                        <CloseIcon />
                    </IconButton>
                </Box>
            </DialogTitle>
            <DialogContent>
                <DialogContentText>
                    <Typography
                        gutterBottom
                        paddingX={desktopMode ? 8 : 4}
                        color='text.secondary'
                        variant='subtitle1'
                        fontStyle='italic'>
                        {t('apply_note')}
                    </Typography>
                    <Typography
                        gutterBottom
                        paddingX={desktopMode ? 4 : 1}
                        paddingTop={2}
                        color='text.primary'
                        variant='body1'>
                        {description}
                    </Typography>
                </DialogContentText>
                <FormGenerator
                    elevated={false}
                    definition={definition}
                    onSubmit={onSubmit} />
            </DialogContent>
        </Dialog>
    );
};
