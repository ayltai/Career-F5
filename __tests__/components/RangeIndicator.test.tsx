import React from 'react';

import { render, } from '../../src/utils/test';
import { RangeIndicator, } from '../../src/components/RangeIndicator';

describe('RangeIndicator', () => {
    it('renders correctly', () => expect(render(
        <RangeIndicator
            color='primary'
            name='Test'
            min={0}
            max={100}
            salaries={{
                min : 25,
                max : 75,
                avg : 50,
            }} />
    )).toMatchSnapshot());
});
