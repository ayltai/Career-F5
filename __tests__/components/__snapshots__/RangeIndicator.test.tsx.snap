// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeIndicator renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-8cva6s-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1t2ey5u-MuiStack-root"
        >
          <div
            class="MuiBox-root css-i9gxme"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-4mz6wg-MuiTypography-root"
            >
              Test
            </p>
          </div>
          <div
            class="MuiBox-root css-2uchni"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-4mz6wg-MuiTypography-root"
            >
              50
            </p>
          </div>
        </div>
        <span
          class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-1i3537d-MuiSlider-root"
        >
          <span
            class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
          />
          <span
            class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
            style="left: 25%; width: 50%;"
          />
          <span
            class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-3nibdl-MuiSlider-thumb"
            data-index="0"
            style="left: 25%;"
          >
            <input
              aria-orientation="horizontal"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="25"
              data-index="0"
              max="100"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="25"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-valueLabel css-1jknfwc-MuiSlider-valueLabel"
            >
              <span
                class="MuiSlider-valueLabelCircle"
              >
                <span
                  class="MuiSlider-valueLabelLabel"
                >
                  25
                </span>
              </span>
            </span>
          </span>
          <span
            class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-3nibdl-MuiSlider-thumb"
            data-index="1"
            style="left: 75%;"
          >
            <input
              aria-orientation="horizontal"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="75"
              data-index="1"
              max="100"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="75"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-valueLabel css-1jknfwc-MuiSlider-valueLabel"
            >
              <span
                class="MuiSlider-valueLabelCircle"
              >
                <span
                  class="MuiSlider-valueLabelLabel"
                >
                  75
                </span>
              </span>
            </span>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-8cva6s-MuiStack-root"
    >
      <div
        class="MuiStack-root css-1t2ey5u-MuiStack-root"
      >
        <div
          class="MuiBox-root css-i9gxme"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-4mz6wg-MuiTypography-root"
          >
            Test
          </p>
        </div>
        <div
          class="MuiBox-root css-2uchni"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-4mz6wg-MuiTypography-root"
          >
            50
          </p>
        </div>
      </div>
      <span
        class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-1i3537d-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style="left: 25%; width: 50%;"
        />
        <span
          class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-3nibdl-MuiSlider-thumb"
          data-index="0"
          style="left: 25%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="25"
            data-index="0"
            max="100"
            min="0"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="25"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabel css-1jknfwc-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                25
              </span>
            </span>
          </span>
        </span>
        <span
          class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-3nibdl-MuiSlider-thumb"
          data-index="1"
          style="left: 75%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="75"
            data-index="1"
            max="100"
            min="0"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="75"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabel css-1jknfwc-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                75
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
